IntegrationTests/TimersTest.js

```diff
@@ -151,12 +164,75 @@ const TimersTest = createReactClass({
     );
   },
 
+  _setTimeout: function(fn: () => void, ms?: number): TimeoutID {
+    const timeoutID = setTimeout(() => {
+      this._timeoutIDs.splice(this._timeoutIDs.indexOf(timeoutID));
+      fn();
+    }, ms);
+    this._timeoutIDs.push(timeoutID);
+    return timeoutID;
+  },
+
+  _setInterval: function(fn: () => void, ms?: number): IntervalID {
+    const intervalID = setInterval(() => {
+      this._intervalIDs.splice(this._intervalIDs.indexOf(intervalID));
```

Comment:
I think this is a bug? When the callback is first called, we'll clear the interval id from `this_intervalIDs`, which means that we'll never clear the interval in `componentWillUnmount`.